params_test.go

```diff
@@ -20,3 +22,66 @@ func TestFormatTime(test *testing.T) {
 	formattedTime := formatTime(time)
 	t.Assert(formattedTime, formatTime(time), "formatTime()")
 }
+
+func TestParamsWithOptions(test *testing.T) {
+	t := &T{test}
+
+	ctx := context.Background()
+	iKey := "asdf"
+	p := &Params{}
```

Comment:
I would say what we want a `New` function for creating params, e.g.
```go
func NewParams(opts ...Option)
```
that way you don't have to call `ApplyOptions` thereafter, the new function can just iterate through them all in one step

Leave the `ApplyOptions` method too, so there's flexibility in when a user decides to add more options

Reply:
Also sets us up nicely for creating any defaults the params should have in the future